x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/ApiKeyBackwardsCompatibilityIT.java

```diff
@@ -67,7 +67,8 @@ public void testCreatingAndUpdatingApiKeys() throws Exception {
                     createException.getMessage(),
                     anyOf(
                         containsString("failed to parse role [my_role]. unexpected field [remote_indices]"),
-                        containsString("remote indices not supported for API keys")
+                        containsString("remote indices not supported for API keys"),
+                        containsString("unknown index privilege")
```

Comment:
Do we ... expect an unknown index privilege message error after introducing a new privilege?

Reply:
The test here pertains to backwards compatibility, specifically for API keys. For API keys, we serialize the role descriptors (as XContent) associated with it (which includes index privileges), and store those as a doc in the security index. The reason the test is failing here is because the test runs on an old version that doesn't have `manage_dlm` defined as a privilege yet. Further up in the test, we are creating a random role descriptor with a set of random index privileges, that sometimes include `manage_dlm` leading to the unknown index privilege failure here.

Just wanted to leave this as context, to save a bit of digging.

In terms of addressing this, for the context of this PR, I think it's all good to keep this as is. I might follow this up next week to make things a little more robust in a mixed-cluster scenario but it's too much of a side quest for the current PR.